server: check all meta records during decommissioning

tbg opened the pull request:
By scanning only the meta2 information, replicas of the meta2 ranges
itself would be ignored. Include meta1 in the scan to address this.

This wasn't released, so no release note has been added. I can add a test
to this if requested, but there wasn't a straightforward way to do so and
this is not the kind of bug that would regress, so it seemed that my time
would better be spent elsewhere.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/27291)
<!-- Reviewable:end -->

bdarnell commented:
I think there were some comments that referred to meta2 that should also be updated.

tschottdorf commented:
Good point, done.

tbg committed:
server: check all meta records during decommissioning

By scanning only the meta2 information, replicas of the meta2 ranges
itself would be ignored. Include meta1 in the scan to address this.

This wasn't released, so no release note has been added.

Release note: None
